exporter/loadbalancingexporter/trace_exporter.go

```diff
@@ -96,54 +106,70 @@ func (e *traceExporterImp) ConsumeTraces(ctx context.Context, td ptrace.Traces)
 }
 
 func (e *traceExporterImp) consumeTrace(ctx context.Context, td ptrace.Traces) error {
-	traceID := traceIDFromTraces(td)
-	if traceID == pcommon.InvalidTraceID() {
-		return errNoTracesInBatch
-	}
-
-	endpoint := e.loadBalancer.Endpoint(traceID)
-	exp, err := e.loadBalancer.Exporter(endpoint)
+	routingIds, err := routingIdentifierFromTraces(td, e.routingKey)
 	if err != nil {
 		return err
 	}
-
-	te, ok := exp.(component.TracesExporter)
-	if !ok {
-		expectType := (*component.TracesExporter)(nil)
-		return fmt.Errorf("expected %T but got %T", expectType, exp)
-	}
-
-	start := time.Now()
-	err = te.ConsumeTraces(ctx, td)
-	duration := time.Since(start)
-	ctx, _ = tag.New(ctx, tag.Upsert(tag.MustNewKey("endpoint"), endpoint))
-
-	if err == nil {
-		sCtx, _ := tag.New(ctx, tag.Upsert(tag.MustNewKey("success"), "true"))
-		stats.Record(sCtx, mBackendLatency.M(duration.Milliseconds()))
-	} else {
-		fCtx, _ := tag.New(ctx, tag.Upsert(tag.MustNewKey("success"), "false"))
-		stats.Record(fCtx, mBackendLatency.M(duration.Milliseconds()))
+	for rid := range routingIds {
+		endpoint := e.loadBalancer.Endpoint([]byte(rid))
+		exp, err := e.loadBalancer.Exporter(endpoint)
+		if err != nil {
+			return err
+		}
+
+		te, ok := exp.(component.TracesExporter)
+		if !ok {
+			expectType := (*component.TracesExporter)(nil)
+			return fmt.Errorf("expected %T but got %T", expectType, exp)
+		}
+
+		start := time.Now()
+		err = te.ConsumeTraces(ctx, td)
+		duration := time.Since(start)
+		ctx, _ = tag.New(ctx, tag.Upsert(tag.MustNewKey("endpoint"), endpoint))
+
+		if err == nil {
+			sCtx, _ := tag.New(ctx, tag.Upsert(tag.MustNewKey("success"), "true"))
+			stats.Record(sCtx, mBackendLatency.M(duration.Milliseconds()))
+		} else {
+			fCtx, _ := tag.New(ctx, tag.Upsert(tag.MustNewKey("success"), "false"))
+			stats.Record(fCtx, mBackendLatency.M(duration.Milliseconds()))
+		}
 	}
-
 	return err
 }
 
-func traceIDFromTraces(td ptrace.Traces) pcommon.TraceID {
+func routingIdentifierFromTraces(td ptrace.Traces, key routingKey) (map[string]bool, error) {
+	ids := make(map[string]bool)
 	rs := td.ResourceSpans()
 	if rs.Len() == 0 {
-		return pcommon.InvalidTraceID()
+		return nil, errors.New("invalid trace id")
 	}
 
 	ils := rs.At(0).ScopeSpans()
 	if ils.Len() == 0 {
-		return pcommon.InvalidTraceID()
+		return nil, errors.New("invalid trace id")
 	}
 
 	spans := ils.At(0).Spans()
 	if spans.Len() == 0 {
-		return pcommon.InvalidTraceID()
+		return nil, errors.New("invalid trace id")
 	}
 
-	return spans.At(0).TraceID()
+	switch key {
+	case svcRouting:
+		for i := 0; i < rs.Len(); i++ {
+			svc, ok := rs.At(i).Resource().Attributes().Get("service.name")
+			if !ok {
+				return nil, errors.New(fmt.Sprintf("unable to get service name"))
+			}
+			ids[svc.StringVal()] = true
+		}
+		return ids, nil
+	case traceIdRouting:
+		tid := spans.At(0).TraceID().Bytes()
+		ids[string(tid[:])] = true
+		return ids, nil
+	}
+	return nil, nil
```

Comment:
nit: Assuming we perform the `routing_key` validation earlier on, _I think_ this line is not reachable since there are only two possible values for `routingKey`? As such, we could simplify this with just a:
```
if key == svcRouting {
    ...
    return ids, nil
}
tid := spans.At(0).TraceID().Bytes()
ids[string(tid[:])] = true
return ids, nil
```